### PR TITLE
[TS] LPS-197703

### PR DESCRIPTION
**Pull request by balazssk**
Motivation
=======
If a JournalArticle gets created with an invalid folderId, the JournalPortlet will not be able to display any articles

Proposed Solution
========
Force a folderId validation through JournalArticleModelValidator

Steps to Verify
========
1. Go to Content & Data > Web Content
2. Add a Basic Web Content and publish
3. Edit the web content
4. Open the browser developer tools and find the element with the id=”_com_liferay_journal_web_portlet_JournalPortlet_folderId"
5. Change the value attribute from 0 to -1 and press enter
6. Press the Save as Draft button
7. An error page appears
8. Go back to Content & Data > Web Content

**Expected behaviour:** The page loads the web content
**Actual behaviour:** The Page does not load any web content

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/validation/JournalArticleModelValidator.java [hunk 1]
```diff
@@ -438,15 +439,15 @@ public ModelValidationResults validateModel(JournalArticle article) {
...
+			String smallImageURL, byte[] smallImageBytes, long smallImageId,
+			int smallImageSource, String content)
 		throws PortalException {
 
-		if (ddmStructureId > 0) {
-			_ddmStructureLocalService.getDDMStructure(ddmStructureId);
-		}
+		_journalFolderLocalService.getFolder(folderId);
+
+		_ddmStructureLocalService.getDDMStructure(ddmStructureId);
```
Comment (jkappler) [inline modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/validation/JournalArticleModelValidator.java:450]: Hi @balazssk, why are removing the condition for checking ddmStructureId? Would it also make sense to have the same condition for folder?

Comment (jkappler) [reply modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/validation/JournalArticleModelValidator.java:450]: Got it @balazssk 👍. Then, can you please also add an integration test for this? I don't want us to break it in the future by mistake.

Thanks!

Comment (balazssk) [reply modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/validation/JournalArticleModelValidator.java:450]: On it

**Comment by balazssk** [comment]
Hi @jkappler ,
You've had a good hunch there, folderId=0 is a special case that should not be the cause of failure.

**Comment by balazssk** [comment]
ci:test:sf

**Comment by jkappler** [comment]
Awesome @balazssk! Thanks for the changes!

**Comment by ealonso** [comment]
ci:forward:force

**Pull request closed without merging**